### PR TITLE
Don't give the supplied Wayland extension filter the choice of supporting extensions that are not optionally supported

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Fixes: #1558

### CODE REVIEW

File: src/miral/wayland_extensions.cpp [hunk 1]
```diff
@@ -267,7 +267,14 @@ void miral::WaylandExtensions::add_extension(Builder const& builder)
...
 void miral::WaylandExtensions::set_filter(miral::WaylandExtensions::Filter const& extension_filter)
 {
-    self->extensions_filter = extension_filter;
+    // Wayland calls the filter for all protocols (not just the optional extensions).
+    // To avoid accidents (like denying base protocols) we only defer to the provided
+    // extension_filter for supported_extensions.
+    self->extensions_filter = [&optional = self->supported_extensions, extension_filter]
+        (Application const& app, char const* protocol)
+        {
+            return optional.count(protocol) == 0 || extension_filter(app, protocol);
```
[the remark this answers is not in the document]
Comment (AlanGriffiths) [reply src/miral/wayland_extensions.cpp:276]: Well, I might be persuaded to `(optional.count(protocol) == 0) || extension_filter(app, protocol)` or `!optional.count(protocol) || extension_filter(app, protocol)` but splitting it over 9 lines seems verbose and that's hard to read.

[the remark this answers is not in the document]
Comment (wmww) [reply src/miral/wayland_extensions.cpp:276]: `return optional.count(protocol) ? extension_filter(app, protocol) : true;`?

[the remark this answers is not in the document]
Comment (AlanGriffiths) [reply src/miral/wayland_extensions.cpp:276]: > return optional.count(protocol) ? extension_filter(app, protocol) : true;

1. Why abuse the ternary `?:` operator when we've got a perfectly good short-circuiting binary `||` operator?
2. You're happy with the implicit integer type conversion to `bool`?

@RAOF, @Saviq any chance of choosing a colour for this bikeshed?

[the remark this answers is not in the document]
Comment (Saviq) [reply src/miral/wayland_extensions.cpp:276]: My vote would be for:
```cpp
(optional.count(protocol) == 0) || extension_filter(app, protocol)
```

**Pull request merged**